openedx/adg/lms/webinars/admin.py

```diff
@@ -22,12 +26,54 @@ class WebinarAdmin(admin.ModelAdmin):
     readonly_fields = ('created_by', 'modified_by',)
     filter_horizontal = ('co_hosts', 'panelists',)
 
+    form = WebinarForm
+
+    def get_fields(self, request, obj=None):
+        """
+        Override `get_fields` to dynamically set fields to be rendered.
+        """
+        fields = super().get_fields(request, obj)
+        if not obj:
+            fields.remove('send_update_emails_to_registrants')
+        return fields
+
     def save_model(self, request, obj, form, change):
+        """
+        Extension of save_model for webinar to set the created_by and modified_by fields.
+        """
         if not change:
             obj.created_by = request.user
         else:
             obj.modified_by = request.user
-        obj.save()
+        super(WebinarAdmin, self).save_model(request, obj, form, change)
+
+    def save_related(self, request, form, formsets, change):
+        """
+        Extension of save_related for webinar to send emails when object is created or modified.
+        """
+        super(WebinarAdmin, self).save_related(request, form, formsets, change)
+
+        webinar = form.instance
+
+        emails_for_event_invite = form.cleaned_data.get('invites_by_email_address', [])
+        if form.cleaned_data.get('invite_platform_users'):
+            emails_for_event_invite += list(
+                User.objects.exclude(email='').values_list('email', flat=True)
+            )
+
+        if change:
+            if form.cleaned_data.get('send_update_emails_to_registrants'):
+                registered_users = list(
+                    webinar.registrations.filter(is_registered=True).values_list('user__email', flat=True)
+                )
+                send_webinar_emails(webinar, list(set(registered_users)), MandrillClient.WEBINAR_UPDATED)
+
+                emails_for_event_invite = remove_duplicate_emails(emails_for_event_invite, registered_users)
```

Comment:
How about making `emails_for_event_invite` a set? it would automatically take care of uniqueness? we won't need to write a separate helper for this

Reply:
It removes the email from emails_for_event_invite that are also present in registered_users list. So, anyone receiving the email for event update, doesn't receive the invitation email.